src/mpi/coll/reduce_local/reduce_local.c

```diff
@@ -61,9 +61,15 @@ int MPIR_Reduce_local(const void *inbuf, void *inoutbuf, MPI_Aint count, MPI_Dat
         if (mpi_errno != MPI_SUCCESS)
             goto fn_exit;
         /* --END ERROR HANDLING-- */
-        /* get the function by indexing into the op table */
-        uop = MPIR_OP_HDL_TO_FN(op);
-        (*uop) ((void *) inbuf, inoutbuf, &count, &datatype);
+        if (MPIR_Typerep_reduce_is_supported(op, datatype)) {
```

Comment:
@nusislam I guess this is just a local (almost constant) check, its performance impact is negligible, right? I don't think we even need to worry about further optimization. Do you think you can run a quick benchmark against the optimum path (your internal branch) to verify?

Reply:
This is a local function call that involves several branches. I can run tests to see how this function call impacts the performance of local reduce.

Reply:
@hzhou - I measured the time for `MPI_Reduce_local()` with host buffers for both yaksa (`seq` backend) vs. host. The times are as follows:
msg size     host(us)    yaksa(us)
1KB             0.88          0.92
32KB           0.23          0.23

The difference is not much.

Reply:
How does `32kb` ran faster than `1kb`? 

Anyway, until we see a reason to optimize, let's not to worry about it. I think for now this is good enough, do you agree?

Reply:
@hzhou - Good catch. Missed a 0 after the decimal point.